Lib/idlelib/run.py

```diff
@@ -474,6 +475,44 @@ def __init__(self, rpchandler):
         self.calltip = calltip.Calltip()
         self.autocomplete = autocomplete.AutoComplete()
 
+        self._wrap_sys_setrecursionlimit()
+
+    def _wrap_sys_setrecursionlimit(self):
+        code = textwrap.dedent("""\
+            import sys
+            from functools import wraps
+
+            # add the delta to the default recursion limit, to compensate
+            sys.setrecursionlimit(sys.getrecursionlimit() + 25)
+
+            @wraps(sys.setrecursionlimit)
+            def setrecursionlimit(*args, **kwargs):
+                # mimic the original sys.setrecursionlimit()'s input handling
+                if kwargs:
+                    raise TypeError(
+                        "setrecursionlimit() takes no keyword arguments")
+                try:
+                    limit, = args
+                except ValueError:
+                    raise TypeError("setrecursionlimit() takes exactly one "
+                                    "argument ({} given)".format(len(args)))
+                if not limit > 0:
+                    raise ValueError(
+                        "recursion limit must be greater or equal than 1")
+
+                return setrecursionlimit.__wrapped__(limit + 25)
+            sys.setrecursionlimit = setrecursionlimit
+
+            @wraps(sys.getrecursionlimit)
+            def getrecursionlimit():
+                return getrecursionlimit.__wrapped__() - 25
+            sys.getrecursionlimit = getrecursionlimit
+
+            del sys
+            del wraps
+            """)
+        exec(code, self.locals)
```

Comment:
I believe that this code should be executed in the run module on import, NOT in the main module.  Calling a module function from main, as I suggested on the issue, should do that.  Then the deletions are not needed.

Reply:
Done, except that call to install_recursionlimit_wrappers is in MyHandler.handle rather than main().  I presume the rationale is that 4 other sys attributes are replaced here.  Good enough.